modules/ve-mw/dm/models/ve.dm.MWTemplateModel.js

```diff
@@ -341,10 +341,17 @@ ve.dm.MWTemplateModel.prototype.serialize = function () {
 		template = { target: this.getTarget(), params: {} },
 		params = this.getParameters();
 
+	if ( !template.originalParams ) {
+		template.originalParams = this.originalData ? Object.keys( this.originalData.params ) : [];
+	}
+
 	for ( name in params ) {
 		if ( name === '' ) {
 			continue;
 		}
+		if ( params[name].getValue() === '' && template.originalParams.indexOf( name ) === -1 ) {
+			continue;
+		}
```

Comment:
Let's add some comments around this block explaining it's our change and why it's needed